src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -203,14 +229,20 @@ object Instance {
     tasks.iterator.map(task => task.instanceId -> task).toMap
 
   // TODO(PODS-BLOCKER) ju remove apply
-  def apply(task: Task): Instance = new Instance(task.taskId.instanceId, task.agentInfo,
-    InstanceState(
-      status = task.status.taskStatus,
-      since = task.status.startedAt.getOrElse(task.status.stagedAt),
-      version = task.version.getOrElse(Timestamp.zero),
-      healthy = None),
-    Map(task.taskId -> task))
-
+  def apply(task: Task): Instance = {
+    def defaultVersion: Timestamp = {
+      // TODO(PODS): fix this
+      log.error("A default Timestamp.zero breaks things!")
+      Timestamp.zero
+    }
+    new Instance(task.taskId.instanceId, task.agentInfo,
+      InstanceState(
+        status = task.status.taskStatus,
+        since = task.status.startedAt.getOrElse(task.status.stagedAt),
+        version = task.version.getOrElse(defaultVersion),
```

Comment:
Yes, Task.Reserved does currently not have a version unfortunately. An instance will always have one though.